simtools/visualization/visualize.py

```diff
@@ -699,3 +703,347 @@ def plot_array(telescopes, rotate_angle=0, show_tel_label=False):
+        instance of corsika.corsika_output.corsikaOutput.
+    log_z: bool
+        if True, the intensity of the color bar is given in logarithmic scale.
+    """
+    return _kernel_plot_2D_photons(
+        corsika_output_instance, "num_photons_per_telescope", log_z=log_z
+    )
+
+
+def _kernel_plot_1D_photons(corsika_output_instance, property_name, log_y=True):
+    """
+    Create the figure of a 1D plot. The parameter `name` indicate which plot. Choices are
+    "counts", "density", "direction".
+
+    Parameters
+    ----------
+    corsika_output_instance: corsika.corsika_output.corsikaOutput
+        instance of corsika.corsika_output.corsikaOutput.
+    property_name: string
+        Name of the quantity. Options are: "wavelength", "counts", "density", "time", "altitude",
+        "num_photons".
+    log_y: bool
+        if True, the intensity of the Y axis is given in logarithmic scale.
+
+    Returns
+    -------
+    list
+        List of figures for the given telescopes.
+
+    Raises
+    ------
+    ValueError
+        if `name` is not allowed.
+    """
+
+    x_label = {
+        "wavelength": "Wavelength (nm)",
+        "counts": "Distance to center (m)",
+        "density": "Distance to center (m)",
+        "time": "Time since 1st interaction (ns)",
+        "altitude": "Altitude of emission (km)",
+        "num_photons": "Number of photons per event",
+    }
+    if property_name not in x_label:
+        msg = "results: status must be one of {}".format(x_label)
+        corsika_output_instance._logger.error(msg)
+        raise ValueError(msg)
+
+    if property_name == "wavelength":
+        edges, hist_values = corsika_output_instance.get_photon_wavelength_distr()
+    elif property_name == "counts":
+        edges, hist_values = corsika_output_instance.get_photon_radial_distr(density=False)
+    elif property_name == "density":
+        edges, hist_values = corsika_output_instance.get_photon_radial_distr(density=True)
+    elif property_name == "time":
+        edges, hist_values = corsika_output_instance.get_photon_time_distr()
+    elif property_name == "altitude":
+        edges, hist_values = corsika_output_instance.get_photon_altitude_distr()
+    elif property_name == "num_photons":
+        edges, hist_values = corsika_output_instance.get_num_photons_distr()
+        edges, hist_values = [edges], [hist_values]
+
+    all_figs = []
+    for i_hist, _ in enumerate(edges):
+        fig, ax = plt.subplots()
+        ax.bar(
+            edges[i_hist][:-1],
+            hist_values[i_hist],
+            align="edge",
+            width=np.abs(np.diff(edges[i_hist])),
+        )
+        ax.set_xlabel(x_label[property_name])
+        ax.set_ylabel("Counts")
+
+        if log_y is True:
+            ax.set_yscale("log")
+        if corsika_output_instance.telescope_indices is None or property_name == "num_photons":
+            fig.savefig("histogram_" + property_name + "_tels.png", bbox_inches="tight")
```

Comment:
Please use f-string everywhere (in particular it is important not to "add" strings as this is very slow).